src/NUnitEngine/nunit.engine/Drivers/NUnit2DriverFactory.cs

```diff
@@ -57,9 +58,9 @@ public bool IsSupportedTestFramework(string assemblyName, Version version)
         /// <param name="assemblyName">The name of the test framework reference</param>
         /// <param name="version">The version of the test framework reference</param>
         /// <returns></returns>
-        public IFrameworkDriver GetDriver(AppDomain domain, string assemblyName, Version version)
+        public IFrameworkDriver GetDriver(AppDomain domain, AssemblyName reference)
         {
-            if (!IsSupportedTestFramework(assemblyName, version))
+            if (!IsSupportedTestFramework(reference))
                 throw new ArgumentException("Invalid framework name", "frameworkAssemblyName");
```

Comment:
Probably a moot point, but if we have been forced into using VS2015 and the newer mono compilers for portable, then most of those C# 6 features work on all frameworks we target except for CF since they are just syntactic sugar. This one would be helpful because it gives us type safety. I know I have fixed a few misnamed arguments in our guard statements :smile: